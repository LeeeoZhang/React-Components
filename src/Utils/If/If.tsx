import * as React from 'react'

interface ifProps {
  children: React.ReactNode,
  isShow: boolean,
}

function If(props: ifProps): boolean | React.ReactNode {

  const { children, isShow } = props

  return isShow && children

}


export default If
