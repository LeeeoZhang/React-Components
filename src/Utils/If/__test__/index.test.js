import * as React from 'react'
import If from '../If'
import renderer from 'react-test-renderer'

test('测试一下', () => {
  const component = renderer.create(
    <If isShow={false}>
      xxx
    </If>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
