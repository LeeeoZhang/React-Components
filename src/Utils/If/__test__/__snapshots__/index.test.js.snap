// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试一下 1`] = `null`;
